src/util/indirect_references.py

```diff
@@ -109,12 +110,75 @@ def indirect_references(cr, bound_only=False):
 
         yield ir
 
+    if version_gte("saas~17.5"):
+        # jsonb company dependent fields
+        # todo: share the table with table_of_model
+        exceptions = dict(
+            line.split()
+            for line in splitlines(
+                """
+            ir.actions.actions          ir_actions
```

Comment:
What if select all jsonb columns of stored m2o fields? You can still hardcode here the exceptions for which the model doesn't match the one from the field with the usual replace (. -> _).

I'm not worried about performance, this is not a common operation and in the context of an upgrade it won't be noticeable. I'm worried about correctness. We should update custom code fields, for which we do not have the code ever during the upgrade.

In the end the most bulletproof approach would be to have a `company_dependent` column in ir_model_fields, but that's a hard decision. But given that now we have many more jsonb columns (translated fields) I'd may be worth to have a way to differentiate them. 

Finally, is it possible to have a translated company dependent field? How does this work in such case?

Reply:
> What if we select all jsonb columns of stored m2o fields?
reduce the joined lines to improve the performance? If so do you think I should UPDATE ir_model_fields store=True in `_convert_field_to_company_dependent` in case the store column is really inconsistent with the ORM?

> Finally, is it possible to have a translated company dependent field? How does this work in such case?
Don't warry it is not supported and will raise warning in ORM

Reply:
> reduce the joined lines to improve the performance? 

So, I just re-read the code and you are doing exactly what I asked for, my bad. The only difference is the check for `store=True`. I think it would be more consistent, if the field is not stored we can ignore the column. It may be some leftover that we better not update.

>  If so do you think I should UPDATE ir_model_fields store=True in _convert_field_to_company_dependent in case the store column is really inconsistent with the ORM?

I think you can, yes, and log an error if the column is not stored. @KangOl wdyt?

> Don't warry it is not supported and will raise warning in ORM

Here is what worries me: the only way to unambiguously differentiate right now what's the purpose of a jsonb column is via code once the fields are _loaded_. That poses problems for upgrades because we never have custom code. Thus having some metadata in ir_model_fields that clearly discriminates about what's the purpose of the column may be worth it. One hypothetical example was translated company dependent fields. Anyway I guess we can think about that once the case comes. Just please have this in mind for the (future) patches in the ORM :)

Reply:
Yes, we could force the `store` column to `true` with a warning if it wasn't.